src/expr/src/relation/mod.rs

```diff
@@ -2952,7 +2953,7 @@ pub struct RowSetFinishing {
     /// Order rows by the given columns.
     pub order_by: Vec<ColumnOrder>,
     /// Include only as many rows (after offset).
-    pub limit: Option<usize>,
+    pub limit: Option<NonNeg<i64>>,
```

Comment:
Is the reason for using `NonNeg<i64>` instead of `usize` or `u64` is because `LIMIT` in SQL only accepts a positive 64 bit signed integer?

Reply:
Yep, precisely. PostgreSQL doesn't have unsigned integers, so they represent limits as a signed 64-bit integer, and reject negative limits.